sbi/inference/snl/snl.py

```diff
@@ -304,44 +309,46 @@ def __call__(
         """
         self.likelihood_nn = likelihood_nn
         self.prior = prior
-        self.observation = observation
+        self.x = x
 
         if mcmc_method in ("slice", "hmc", "nuts"):
             return self.pyro_potential
         else:
             return self.np_potential
 
-    def np_potential(self, parameters: np.array) -> Union[Tensor, float]:
-        """Return posterior log prob. of parameters."
+    def np_potential(self, theta: np.array) -> Union[Tensor, float]:
+        """Return posterior log prob. of theta p(theta|x)."
         
         Args:
-            parameters: parameter vector, batch dimension 1
+            theta: parameters theta, batch dimension 1
         
         Returns:
-            Posterior log probability of the parameters, -Inf if impossible under prior.
+            Posterior log probability of the theta, -Inf if impossible under prior.
         """
-        parameters = torch.as_tensor(parameters, dtype=torch.float32)
+        theta = torch.as_tensor(theta, dtype=torch.float32)
         log_likelihood = self.likelihood_nn.log_prob(
-            inputs=self.observation.reshape(1, -1), context=parameters.reshape(1, -1)
+            inputs=self.x.reshape(1, -1), context=theta.reshape(1, -1)
```

Comment:
`x_o` has always shape `(1, x_event_shape)`. 
`theta` shape depends on how it was sampled from the prior. we cant control that in the input checking. it could be `(theta_event_shape)` or `(batch_size, theta_event_shape)`.